### PR TITLE
few more versions of sampling

### PR DESCRIPTION
**Pull request by melvinebenezer**
## Sampling of logits in Ring Attention
- Greedy
- top_k
- top_p

### CODE REVIEW

**Comment by melvinebenezer** [comment]
The sampling is done, can moved the logit generation to decoding.py to avoid other's changes

**Review by andreaskoepf — APPROVED** [review]
